### PR TITLE
Freethreading documentation

### PR DESCRIPTION
**Pull request by da-woods**
Just a fairly basic summary of the current status and some known problems.

### CODE REVIEW

File: docs/src/userguide/freethreading.rst [hunk 1]
```diff
@@ -0,0 +1,108 @@
+*************
+Freethreading
```
Comment (lysnikolaou) [inline docs/src/userguide/freethreading.rst:2]: Let's use the term like it's used in the Python docs.

```suggestion
Free threading
```

File: docs/src/userguide/freethreading.rst [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+*************
+
+The Freethreading (sometimes known as "nogil") build of Python is an
```
Comment (lysnikolaou) [inline docs/src/userguide/freethreading.rst:5]: Same comment as above. Ditto in other places.

```suggestion
The free-threaded (sometimes known as "nogil") build of Python is an
```

File: docs/src/userguide/freethreading.rst [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+experimental mode available from Python 3.13 onwards. It aims to
+remove the "Global Interpreter Lock" and allow multiple Python threads to
```
Comment (lysnikolaou) [inline docs/src/userguide/freethreading.rst:7]:
```suggestion
disable the "Global Interpreter Lock" and allow multiple Python threads to
```

**Review by lysnikolaou — COMMENTED** [review]
Thanks @da-woods! I did a first pass. Looks good in general, I only left some nitty comments.

**Comment by da-woods** [comment]
Thanks @lysnikolaou - updated it as you suggest to match Python's style.

**Pull request merged**